### PR TITLE
p2p: Return a max of 1000 addrs from CAddrMan::GetAddr

### PR DESCRIPTION
**Pull request by Empact**
This is the documented max size for an addr response, and was separately being
limited in PushAddress to 1000 via MAX_ADDR_TO_SEND. No point in selecting a
too-large random set then filtering it randomly down to a smaller set.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
I think addrman would return a larger list that is node-agnostic and then net would filter that list taking into account the node state, no?

File: src/addrman.h [hunk 1]
```diff
@@ -172,7 +172,7 @@ class CAddrInfo : public CAddress
 #define ADDRMAN_GETADDR_MAX_PCT 23
 
 //! the maximum number of nodes to return in a getaddr call
-#define ADDRMAN_GETADDR_MAX 2500
+#define ADDRMAN_GETADDR_MAX 1000
```
Comment (practicalswift) [inline src/addrman.h]: When we're editing this file – what about skipping `#define` and using proper constants instead? :-)

**Comment by gmaxwell** [comment]
I thought the two layer setup was so that multiple probes would get more addresses but not enumerate the whole table.

**Comment by gmaxwell** [comment]
@sipa ^

File: src/net_processing.cpp [hunk 2]
```diff
@@ -1757,7 +1757,7 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
             }
 
             // Get recent addresses
-            if (pfrom->fOneShot || pfrom->nVersion >= CADDR_TIME_VERSION || connman->GetAddressCount() < 1000)
+            if (pfrom->fOneShot || pfrom->nVersion >= CADDR_TIME_VERSION || connman->GetAddressCount() < ADDRMAN_GETADDR_MAX)
```
Comment (laanwj) [inline src/net_processing.cpp]: I like changing these to a constant, however "maximum number of addresses in GETADDR" is a P2P layer constant, I don't think it should use addrman's internal constant.

**Comment by MarcoFalke** [comment]
@Empact `PushAddr` will return early when `addrKnown.contains` it. So when `vAddr` is of size 1000, it might very well send less than 1000 addresses back in the response, even though there are plenty to send. No?

**Comment by Empact** [comment]
@MarcoFalke yep that sounds like the explanation. Thanks!

**Pull request closed without merging**